### ISSUE TITLE
Feature: Symlink/Hardlink button does not allow specifying which file is kept as original / can't seem to change order

### ISSUE THREAD

**Issue by BGrogan** [qarmin/czkawka#903]
"Only works when at least two results in a group are selected.
First is unchanged and second and later are symlinked to first."

Given these files of the same image with different resolutions:
Desktop\Image1_medium.jpg
Desktop\Image1_thumbnail.jpg
Images\Source\Image1_verylarge.jpg

Image1_verylarge.jpg is the file we generally want to keep. However, I think the list is currently alphabetically sorted (?). I assume it will use whatever appears first in the results (Desktop\Image1_medium.jpg) as the main image with the others being removed and linking to that. (How this is handled isn't extremely clear via the tooltip.)

An option to specify/clarify "Keep the file with the highest resolution" OR "Keep the images in (Images\Source) directory" would help this issue.

"Keep the images in (Images\Source) directory" is a useful different situation if, for example, there is a series of edited images that are upscaled to technically be larger size / dimensions, but is compressed and a generally lower quality for web. 

A similar, new feature branching off of this discussion: "Replace all similar files with (the largest filesize OR the file ending in ".raw" OR the largest image dimensions)" 
This feature could be used to first make sure all places are updated with the highest quality picture before deleting a source file as I mentioned above. 

This would also be useful in cases where there are multiple places we want duplicate files. For example, Images\All\Images\Wedding\All\Image1_verylarge.jpg AND a specific directory with a subset like Images\Wedding\Dance\Image1_verylarge.jpg

It would also be generally nice to move these files to trash instead of them being removed because there are frequent cascading problems in general when automating the process of linking files.

**Comment by flansuse** [comment]
This is a gigantic oversight for a crucial feature. 

It applies to hardlinks as well.

In fact, why not just convert the **_UNSELECTED_** items into symlinks? Whatever is selected is treated as the "original" file, and the unselected items are all converted into symlinks.

What about for groups where you select more than one item? Just display a warning message and ignore those groups!

Tada!

**Comment by BGrogan** [comment]
Yes, that would be a nice way of handling it with how the program currently works.

I think you'd probably want to allow only 1 selection and use that as the master file while everything else becomes a hard/symlink though. That seems more intuitive as a planned feature.

To expand on these issues, this may be a new topic but it's somewhat related. Tiered conditions would help with a lot of this. 

Condition:
1a. Never delete the largest file unless there are multiple copies
1b. Never delete the oldest file unless there are multiple copies
2a. Never delete files from /source/raw/
3a. Prefer deleting files in /backup/2023-March/

Tier 1 rules take priority over everything else. In the event the oldest result is not also the largest result, both will be kept. 

Tier 2 rules happen next, and let's say we see the largest file is in /backup/2023-March/ but for some reason isn't in /source/raw which has just a thumbnail version of the image. Neither file will be deleted and I will need to manually investigate (this is good, something weird probably happened).

Tier 3 rules next. For an image that exists in the raw folder AND the backup folder, I want to delete the backup copy as my rules are set up. Since this image passes the above 3 rules, we know when we delete it, it will:
- have a copy elsewhere (exists in /raw/ and /backup/ ) and thus not delete the only copy
- Ensure the oldest copy still exists so I can continue to sort by date later
- With the first bullet point, we it will be unable to delete the /raw/ copy so I don't need to check which one it selected for deletion
- With the third bullet point, I am doubly sure it is deleting the backup copy now. I also know if I include a "pictures-to-email-to-bride" folder that I moved (not copied) the picture out of the raw folder from then it's going to fall back and delete the backup copy instead of the real folder I'm worried about



Currently, as a workaround to this (for duplicate images in particular), I have to manually save the output selection text file and then bring it into Excel to create multiple filters:
(Knowing every item in the list is a duplicate with at least 2 images I can do the following--)
1. Write a function to get the current original row (=row()) into column B. Then, copy this column and paste as values. Do not leave row values as formulas because multiple runs of deletions could interfere cause confusion
3. Filter the list for everything containing "/backup/march-2023/" in the string. (Luckily this works for me, but if my backup folder was called 032023 for example, this could unintentionally include images that have that as a randomly generated number.)
4. Write a function to get the current row and subtract the above item in this filtered list from it (=row()-$B2) into Column C. This will let me know if the item above this row is related and becomes critical in not deleting unique copies if all matching images only exist in the backup path and nowhere else
5. Filter column C. DO NOT INCLUDE values of 1 because this would mean the item directly above this is included and in the event all items are in the backup directory, they would all appear here. 
_NOTE: Step 5 isn't perfect because the order of results is random so the first listed result if it appears here could have a value of 3 referring to a separate grouping of duplicated images, and this could be the highest resolution copy, and this high res could only exist in the backup folder. It will get marked for deletion with this method which is why it isn't perfect._
6. Copy the A Column and paste into notepad++
7. Regex replace ` - .*` with nothing
8. Replace `\n` with `"\ndel "` check/fix the first and last lines 
9. Save as duplicatedelete.bat (windows)
10. run script

Lots of potential for error with this though. It would just be nice if this functionality was built in.

**Comment by flansuse** [comment]
> "I think you'd probably want to allow only 1 selection and use that as the master file while everything else becomes a hard/symlink though. That seems more intuitive as a planned feature."

While that makes sense, and I agree, the reason I mentioned "convert the **selected** items into symlinks" is because for all other tasks, it is the **selected** items that are _deleted_.

So basically if you select a bunch of duplicate files to be deleted, but then after going through this long list you change your mind and decide "Instead of deleting these, I'll just convert them to symlinks." So you'll have to invert your selection, and it might come off as counter-intuitive compared to all other tasks.

Symlinks are trickier than hardlinks or deletions, since only symlinks need an "original" file to link from.

**Comment by thomas725** [comment]
well I wouldn't go as far as calling it unusable, I've used it several times and was happy with it, but now I've stumbled upon a case where the first item is not the one I want to be the original where the other's now link to instead, and the UI doesn't have any option to resort or choose the symlink master item that's kept. which makes it unusable in cases where the decision which one should be master is important and differs from the first one in the groups list.
